curr_scraper/apps/api_caller/call_logic.py

```diff
@@ -0,0 +1,54 @@
+import json
+import os
+from datetime import datetime
+
+import pytz
+import requests
+
+from api_caller import models
+
+
+def call_api():
+    oe_url = os.environ.get('OPEN_EXCHANGE_URL')
+    oe_app_id = os.environ.get('OPEN_EXCHANGE_APP_ID')
+    oe_complete_url = f'{oe_url}/latest.json?app_id={oe_app_id}'
+    try:
+        response = requests.get(oe_complete_url)
+        json_data = json.loads(response.text)
+        base = json_data["base"]
+    except KeyError:
```

Comment:
You capture a wide range of operations in try - except block while covering only basic exception `KeyError` which is mostly dictionary related for `base = json_data["base"]`.

`requests` is HTTP protocol library which can raise ton of different exceptions like ConnectionError, Timeout, HTTPError etc. https://requests.readthedocs.io/en/latest/user/quickstart/#errors-and-exceptions
`JSON` library also has plenty of exceptions which should be handled separately

Please remove lines 16:17 from the try block and add additional exception handling at least for 1 exception each.